src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunction.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Drawing;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    internal class JsonFunctionImpl : JsonFunction, IAsyncTexlFunction4
+    {
+        public Task<FormulaValue> InvokeAsync(TimeZoneInfo timezoneInfo, IRContext irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            return Task.FromResult(new JsonProcessing(timezoneInfo, irContext, args).Process());
+        }
+
+        internal class JsonProcessing
+        {
+            private JsonFlags _flags;
+            private readonly FormulaValue[] _args;
+            private readonly IRContext _irContext;
+            private readonly TimeZoneInfo _timeZoneInfo;
+
+            internal JsonProcessing(TimeZoneInfo timezoneInfo, IRContext irContext, FormulaValue[] args)
+            {
+                _args = args;
+                _irContext = irContext;
+                _timeZoneInfo = timezoneInfo;
+            }
+
+            internal FormulaValue Process()
+            {
+                GetFlags();
+
+                if (_flags == null || HasUnsupportedType(_args[0].Type._type, out _, out _))
+                {
+                    return CommonErrors.GenericInvalidArgument(_irContext);
+                }
+
+                using MemoryStream memoryStream = new MemoryStream();
+                using Utf8JsonWriter writer = new Utf8JsonWriter(memoryStream, new JsonWriterOptions() { Indented = _flags.IndentFour });
+
+                WriteToJson(writer, _args[0]);
+                writer.Flush();
+
+                string json = Encoding.UTF8.GetString(memoryStream.ToArray()).Replace(@"\u0022", @"\""").Replace(@"\u0027", @"'");
```

Comment:
>Replace(@"\u0022", @"\""").Replace(@"\u0027", @"'") [](http://example.com/codeflow?start=78&length=51)

what's this doing?

Reply:
Utf8JsonWriter is over-encoding some characters.
We might want to refine that, either in this PR or in another

Reply:
There is an easier way to do that and it's well hidden in the doc - [https://learn.microsoft.com/en-us/dotnet/api/system.text.encodings.web.javascriptencoder.unsaferelaxedjsonescaping?view=net-7.0](https://learn.microsoft.com/en-us/dotnet/api/system.text.encodings.web.javascriptencoder.unsaferelaxedjsonescaping?view=net-7.0)